### ISSUE TITLE
Uncaught error in Spotify

### ISSUE THREAD

**Issue by gcampax** [stanford-oval/thingpedia-common-devices#292]
```
====
# main
#! timestamp: 2021-03-09T03:14:37.239Z
U: play the elevator playlist.
UT: $dialogue @org.thingpedia.dialogue.transaction.execute;
UT: @com.spotify.playlist() filter contains(artists, null^^com.spotify:artist("elevator")) => @com.spotify.play(playable=id);
C: $dialogue @org.thingpedia.dialogue.transaction.execute;
C: @com.spotify(id="com.spotify-gcampax"^^tt:device_id("Spotify Account of gcampax")).playlist() filter contains(artists, "spotify:artist:2JmhnB4HLAf3PjqmikkJrC"^^com.spotify:artist("Elevator Jay")) => @com.spotify(id="com.spotify-gcampax"^^tt:device_id("Spotify Account of gcampax")).play(playable=id)
C: #[results=[]]
C: #[error="Cannot read property 'some' of undefined"];
#! timestamp: 2021-03-09T03:14:39.308Z
A: Sorry, Cannot read property 'some' of undefined.
AT: $dialogue @org.thingpedia.dialogue.transaction.sys_action_error;
```

Logs:
```
Failed to invoke query TypeError: Cannot read property 'some' of undefined
    at Object.contains (/opt/almond/node_modules/thingtalk/dist/builtin/primitive_ops.js:231:14)
    at eval (eval at compile (/opt/almond/node_modules/thingtalk/dist/compiler/jsir.js:794:46), <anonymous>:57:29)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async RuleExecutor._ruleThread (/opt/almond/node_modules/genie-toolkit/dist/lib/engine/apps/rule_executor.js:62:13)
```

**Comment by rhulkb27** [comment]
Hello @gcampax, may I try working on this issue?

**Comment by gcampax** [comment]
Hey @rhulkb27! Sure, always appreciate new contributors.
I think the bug explanation should be straightforward, but feel free to ask any question you might have. Once you have the fix, open a PR and I will review it.

**Comment by rhulkb27** [comment]
Hi @gcampax just wanted to keep you updated, I was able to repro the issue here:
![image](https://user-images.githubusercontent.com/47308920/113537413-89d07700-958d-11eb-8e95-0527807b6f37.png)
Here was the error output from my terminal:
![image](https://user-images.githubusercontent.com/47308920/113537397-7c1af180-958d-11eb-9853-3ae9dbba1dc5.png)
I suspect the issue is that the playlist requires spotify premium to play. I'll look into fixing this bug now.

**Comment by gcampax** [comment]
Hey, indeed playing anything requires Spotify premium, but I don't think this was the issue here. I have premium, and also we should handle that case separately. Check for the code, you'll find an error message for premium_required or similar. I think the issue here is different, and has to do with playing a playlist that is not found in the list (some code receives an empty list or similar).

**Comment by rhulkb27** [comment]
Hello @gcampax, as we had discussed on discord, this bug seems to have resolved itself. I created my own elevator playlist:
![image](https://user-images.githubusercontent.com/47308920/115167012-b34bd100-a06a-11eb-9be3-b730bba825cb.png). Additionally, this playlist is empty just as your playlist was empty. Instead of giving an error like before, this was the response:
![image](https://user-images.githubusercontent.com/47308920/115167148-353bfa00-a06b-11eb-9aea-e895c42b58c7.png)
The original error message no longer appears.
